### ISSUE TITLE
Pigeon classify error: error loading reference annotations for reference; vector::reserve

### ISSUE THREAD

**Issue by SidG13** [PacificBiosciences/pbbioconda#715]
Hi,

I'm trying to figure out exactly what's wrong with my GTF when running `pigeon classify`, because it seemingly processes the first 7 chromosomes correctly, but something is wrong with the 8th. I've isolated the problem to any chromosome named "scaffold-mi*", where the "*" represents a number. But the GTF lines are identical in structure across the GTF, so I'm completed lost as to why this is happening.

Here's a screenshot of the problematic lines. 
<img width="2266" alt="image" src="https://github.com/user-attachments/assets/ac1fdf8b-1b21-45c1-b303-ec5bd7db802a">

Basically, pigeon classify works un scaffold-ma1 to scaffold-ma7, then drops  the following error:

`| 20240916 20:18:14.118 | FATAL | pigeon classify ERROR: error loading reference annotations for reference: scaffold-mi1
vector::reserve`. 

The original GTF file sorted and indexed fine using `pigeon prepare`, so no issues there. The GTF was formatted as a standard 9 column file as well, with gene_id, transcript_id and gene_name in the 9th column.

Any guidance is appreciated!

**Comment by boweny920** [comment]
I ran into the same issue. Have you found a solution by chance? Thanks! 

> Hi,
> 
> I'm trying to figure out exactly what's wrong with my GTF when running `pigeon classify`, because it seemingly processes the first 7 chromosomes correctly, but something is wrong with the 8th. I've isolated the problem to any chromosome named "scaffold-mi*", where the "*" represents a number (when I remove scaffold-mi1 from the GTF, it fails at scaffold-mi2). But the GTF lines are identical in structure across the GTF, so I'm completed lost as to why this is happening. This is a GTF [specifically modified to work with the IsoSeq pipeline](https://isoseq.how/classification/pigeon-annotation.html), but otherwise I've been using the original GTF for a long time for many other applications, and the chromosomes do exist in the genome fasta, names match.
> 
> The GTF file sorted and indexed fine using `pigeon prepare`, so no issues there. The GTF was formatted as a standard 9 column file as well, with gene_id, transcript_id and gene_name in the 9th column.
> 
> Here's a screenshot of the problematic lines. <img alt="image" width="2000" src="https://private-user-images.githubusercontent.com/36648847/367917764-ac1fdf8b-1b21-45c1-b303-ec5bd7db802a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ1Mzk0MjEsIm5iZiI6MTczNDUzOTEyMSwicGF0aCI6Ii8zNjY0ODg0Ny8zNjc5MTc3NjQtYWMxZmRmOGItMWIyMS00NWMxLWIzMDMtZWM1YmQ3ZGI4MDJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE4VDE2MjUyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0ZjA1MjVlYWM2M2E5MDc3NDcyYjMzNTFiY2UwMWI3NjkwZDAwOWI4YjdiMGRmMmM2NWU3ZDhhZjcxMzEyNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.r6XTscPwSab92xhM500z6tHEoZMgNe64K4jR5o1UsGI">
> 
> The exact command structure I ran is:
> 
> `pigeon classify sample.collapsed.sorted.gff sample.reference.pigeon.sorted.gtf genome.fa`
> 
> Basically, pigeon classify works on seven chromosomes, then drops the following error:
> 
> `| 20240916 20:18:14.118 | FATAL | pigeon classify ERROR: error loading reference annotations for reference: scaffold-mi1 vector::reserve`.
> 
> Separately, I used `agat_convert_sp_gff2gtf.pl` to check for errors in formatting. The GTF produced by AGAT also had the same issue at scaffold-mi1.
> 
> Any guidance is appreciated!
> 
> I've attached my .pgi GTF reference file in case that may help. [CroVir_rnd1.all.maker.final.homologIDs.updatedNov2019_with_myos_geneidmod_pigeon.sorted.gtf.pgi.txt](https://github.com/user-attachments/files/17019617/CroVir_rnd1.all.maker.final.homologIDs.updatedNov2019_with_myos_geneidmod_pigeon.sorted.gtf.pgi.txt)
> 
> Some version details:
> 
> ```
> pigeon --version
> pigeon 1.2.0 (commit -v1.2.0)
> 
> Using:
>   pbbam     : 2.5.0 (commit v2.5.0)
>   pbcopper  : 2.4.0 (commit v2.4.0)
>   boost     : 1.81
>   htslib    : 1.17
>   zlib      : 1.2.13
> ```

**Comment by jmattick** [comment]
Hi @SidG13 and @boweny920, 
Thanks for bringing this to our attention. Could either of you provide a minimal reproducible example data showing the issue? That will help us diagnose the problem.

**Comment by EternalEnvy10** [comment]
Hi, I also have this issue. And when I extract only one chromosome from gtf and genome.fa for testing, this error still occurs
